rust/actyx/ax-aql/src/language/types.rs

```diff
@@ -10,28 +10,28 @@ use once_cell::sync::Lazy;
 use pest::pratt_parser::{Assoc, Op, PrattParser};
 use std::sync::Arc;
 
-#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Hash)]
 pub enum Type {
     Atom(TypeAtom),
     Union(Arc<(Type, Type)>),
     Intersection(Arc<(Type, Type)>),
     Array(Arc<Type>),
     Dict(Arc<Type>),
+    Tuple(NonEmptyVec<Type>),
+    Record(NonEmptyVec<(Label, Type)>),
```

Comment:
shouldn’t this require changes in the `impl Arbitrary`?

Reply:
Yes, this is still a WIP and that part doesn't compile yet. I need to figure out the whole `arb!` macro